cls/SourceControl/Git/DiscardState.cls

```diff
@@ -0,0 +1,140 @@
+Class SourceControl.Git.DiscardState Extends (%Persistent, %JSON.Adaptor)
+{
+
+Property FullExternalName As %String(MAXLEN = "") [ Required ];
+
+Property Name As %String [ Required ];
+
+Property Contents As %Stream.GlobalCharacter(LOCATION = "^SourceControl.Git.DiscardS");
+
+Property Username As %String [ Required ];
+
+Property Branch As %String [ Required ];
+
+Property Timestamp As %TimeStamp [ Required ];
+
+Property ExternalFile As %Boolean [ Required ];
+
+Index BranchMap On Branch [ Type = bitmap ];
+
+Method RestoreToFileTree()
+{
+    // Make sure directory for file exists
+    set dir = ##class(%File).GetDirectory(..FullExternalName)
+    if ('##class(%File).DirectoryExists(dir)) {
+        do ##class(%File).CreateDirectoryChain(dir)
+    }
+    
+    // Recreate File
+    set fileStream = ##class(%Stream.FileCharacter).%New()
+    set fileStream.Filename = ..FullExternalName
+    $$$ThrowOnError(fileStream.CopyFrom(..Contents))
+    $$$ThrowOnError(fileStream.%Save())
+
+    // Add file to source-control / IRIS
+    if '..ExternalFile {
+        do ##class(SourceControl.Git.Utils).ImportItem(..Name, 1, 1)
```

Comment:
We should compile after importing. This is especially important for productions because if the class isn't compiled, we won't see anything changed in the interoperability portal.